app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt

```diff
@@ -175,40 +160,53 @@ class MyWatchFaceDigital : CanvasWatchFaceService() {
                 WatchFaceService.TAP_TYPE_TOUCH_CANCEL -> {
                     // The user has started a different gesture or otherwise cancelled the tap.
                 }
-                WatchFaceService.TAP_TYPE_TAP ->
+                WatchFaceService.TAP_TYPE_TAP -> {
                     // The user has completed the tap gesture.
-                    // TODO: Add code to handle the tap gesture.
-                    Toast.makeText(applicationContext, R.string.message, Toast.LENGTH_SHORT)
-                        .show()
+                    startActivity(Intent(this@MyDigitalWatchFace,MainActivity::class.java))
+                }
+
             }
             invalidate()
         }
 
         override fun onDraw(canvas: Canvas, bounds: Rect) {
```

Comment:
Please don't create any instances inside on draw or if you do make sure it's created only once, onDraw is called very often so it's bad practice to anything except drawing

Reply:
Not getting comment. can you please beat explain ? regarding background & time set ?